06-TimeManagement/omod6_kern_time.c

```diff
@@ -127,30 +132,43 @@ static ssize_t ktime_store(struct class *cl,
 			const char *buf, size_t count)
 {
 	int interval = 0;
+	char *cp;
 	int n;
 
-	n = sscanf(buf, "%d %s", &interval, LogMessage);
-	pr_info("omod6 store (%d, \'%s\')\n", interval, LogMessage);
-	if (n >= 1) {
-		TimerInterval = interval;
-		// start, stop or reconfig timer
-		if (interval == 0) {
-			if (timer_pending(&MyTimer)) {
-				del_timer(&MyTimer);
-				pr_info("omod6 Timer disabled\n");
+	if (sscanf(buf, "%d", &interval) == 1) {
+		cp = strchr(buf, ' ');
+		if (cp) {
+			while (*cp == ' ')
+				cp++;
+			for (n = 0; n < sizeof(LogMessage) - 1; n++) {
```

Comment:
What would be if user enter the wrong/garbage string? Have you tested this case?

Reply:
No, I haven't tested this case. But as for this code, it's made a priory safe by design. It get only limited number of chars in message string and protected against strange ascii codes, e.g. CR/LF, ESC, etc. Which are unwanted in system log. All other ascii chars are OK. This way of string copy  was selected after I realized that store gets the string from user space ended this '\n'. It was a piece of bad news for me.